test/wdio_functional/page_objects/management/index_pattern_page.js

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import BasePage from '../common/base_page';
+import { createTestSubjectSelectorCss } from '../../helpers/helpers';
+
+export default class IndexPatternPage extends BasePage {
+  constructor(driver) {
+    super(driver);
+    this.driver = driver;
+    this.createIndexPatternHeaderSelector = 'h1.euiTitle';
+    this.createIndexPatternNameInputSelector = createTestSubjectSelectorCss('createIndexPatternNameInput');
+    this.createIndexPatternStep2ButtonSelector = createTestSubjectSelectorCss('createIndexPatternGoToStep2Button');
+    this.createIndexPatternCreateButtonSelector = createTestSubjectSelectorCss('createIndexPatternCreateButton');
+    this.createIndexPatternTimeFieldSelectSelector = createTestSubjectSelectorCss('createIndexPatternTimeFieldSelect');
+    this.indexPatternsLinksSelector = createTestSubjectSelectorCss('indexPatternLink');
+    this.indexPatternTitleSelector = createTestSubjectSelectorCss('indexPatternTitle');
+    this.init();
+  }
+  init() {
+    expect('Create index pattern').toBe(this.getElementText(this.createIndexPatternHeaderSelector));
+    this.driver.waitUntil(() => {
+      return this.title === 'Kibana';
+    });
```

Comment:
I think I understand that navigating from mgmt page to index patterns page, causes this constructor to run, which calls init(), and init() is checking that we're on the page based on the title.  But I had to spend a minute looking around to figure that out.  I probably would have put this check in the management_page.navigateToIndexPatterns() method.  But maybe there's a reason I'm not thinking of.

Reply:
I could understand the reasoning here. I guess it could go either way. I am flexible though.